src/main/java/strategies/impl/AnyStrategyImpl.java

```diff
@@ -0,0 +1,27 @@
+package strategies.impl;
+
+import strategies.SearchStrategy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AnyStrategyImpl implements SearchStrategy {
+
+    @Override
+    public List<String> search(List<String> peoples, Map<String, List<Integer>> linesIndexesByWord, String[] queryWords) {
+        List<Integer> wordIndexes = Arrays.stream(queryWords)
+                .flatMap(word -> linesIndexesByWord.getOrDefault(word, new ArrayList<>()).stream())
+                .distinct()
+                .toList();
```

Comment:
This logic is repeated in the `NoneStrategyImpl`, so you can move it to the default method in the `SearchStrategy` interface and override behavior for `AllStrategyImpl`.

Reply:
this logic replaced to SearchStrategy